app.go

```diff
@@ -618,9 +667,16 @@ func (app *App) provide(p provide) {
 
 	switch {
 	case p.IsSupply:
-		app.logger.PrintSupply(constructor)
+		for _, rtype := range fxreflect.ReturnTypes(constructor) {
+			fxlog.Info("supplying", fxlog.Field{Key: "constructor", Value: rtype}).Write(app.log)
+		}
 	default:
-		app.logger.PrintProvide(constructor)
+		for _, rtype := range fxreflect.ReturnTypes(constructor) {
+			fxlog.Info("providing",
+				fxlog.Field{Key: "returnValue", Value: rtype},
```

Comment:
this is a type, not a value. returnType or type.

```suggestion
				fxlog.Field{Key: "type", Value: rtype},
```